techmatters/flex-plugins
# CHI-2763 Resolve In-Progress Contacts: Close Tasks and Finalize Contacts

## Description
- This PR introduces `Save and End` button, responsible for saving unresolved contacts left in progress. 
- A new permission `canEditInProgressContact` permission is added to control the Save and End button
- Refactor the In-Progress Banner with the 'Save and End' Button and display a dialog for saving the completing active calls and saving contacts
- Resolved Banner: Displays a blue banner when the contact is saved.

### Checklist
- [x] Corresponding issue has been [opened](https://tech-matters.atlassian.net/browse/CHI-2763)

### Other Related Issues

None

### Verification steps
- Ensure that this branch is deployed: https://github.com/techmatters/serverless/pull/698
- Start Flex(as a supervisor or owner) and start a call / chat in any of the channels
- Once the task is assigned, access the unfinalized contact via search with the In Progress banner
- You should see `Save and End.` Click on it and check that the behavior is aligned with the acceptance criteria.


### AFTER YOU MERGE

1. Cut a release tag using the Github workflow. Wait for it to complete and notify in the #aselo-deploys Slack channel.
2. Comment on the ticket with the release tag version AND any additional instructions required to configure an environment to test the changes.
3. Only then move the ticket into the QA column in JIRA

You are responsible for ensuring the above steps are completed. If you move a ticket into QA without advising what version to test, the QA team will assume the latest tag has the changes. If it does not, the following confusion is on you! :-P
